records/2024-02-06.AWS-vpn.md

```diff
@@ -0,0 +1,95 @@
+# AWS VPN for GC Notify Infrastructure
+Date: 2024-02-06
+
+## Status
+
+**DRAFT**
+
+## Context
+
+GC Notify uses AWS as its cloud provider. Currently we do not have a standardized way of accessing and maintaining the infrastructure and deployments of GC Notify. Administration endpoints are either public facing (secured by SSO) or require jump boxing into a temporary Kubernetes pod and using a convoluted port-forwarding scheme. 
+
+In order to increase security, standardize access, and comply with GoC standards such as ITSG-22 and ITSG-33, a VPN service should be added to GC Notify's intfrastructure.
+
+## Restrictions
+
+Since we are using AWS, we are restricted to either AWS provided services or self-hosted solutions
+
+## Options
+
+### AWS VPN
+
+AWS provides a first party [VPN solution](https://aws.amazon.com/vpn/) that can be leveraged and implemented via Terraform. It is based on OpenVPN.
+
+#### Pros
+
+- Can be managed in Terraform
+- Integrates directly with AWS CDS Single-Sign-On
+- VPN Client Is Cross Platform
+- Managed Service Requires Little Administration
+- The VPN endpoint protection is done by AWS
+
+#### Cons
+
+- Comparitively Expensive
+- Cloud specific means that we could not directly migrate this to another Provider
+
+#### Cost
+
+AWS charges per hour based on the number of subnet associations. The GC Notify Infrastructure would require associations with three private subnets. This results in a monthly charge of approximately $200 per month per AWS account. Given that GC Notify has 3 accounts (Dev/Staging/Prod), that results in $600 per month, or approximately $7200 per year for all three environments.
```

Comment:
Perhaps a bit late to the discussion (I can see the conclusion to use AWS managed VPN solution, mainly driven by policy compliance), just want to point out that in oder to properly cost justify a security control, it has to be weighed against the value of the asset being protected. In other words, is $7200 a cost-effective safeguard to mitigate the risk of a breach? As already pointed out in discussions below, going with a non-managed VPN solution does not actually cost $0 (administrative overhead, infrastructure costs).

It is also unfortunately not straight-forward to determine the cost-effectiveness of both approaches because it is difficult (if not impossible) to assign a dollar value to Notify (i.e. the amount of loss in case of a breach). Suffice to say, it is a high value asset. If cost is the only consideration (policy compliance is not in the picture), the decision needs to be made by the party accountable for Notify.

Reply:
@cgye Thanks for the comment. We could include this weighted point of view in the rationale for the decision. Just to be clear though: do you think it is best then to bring this up to our executive director?

Reply:
It's beneficial to get input from Sr management (executive director) because as business owners, they usually have a better financial perspective to make the right decisions. It will also keep them informed of the efforts to improve security posture of the organization. We want to make sure security controls put in place are cost-effective and are aligned with the goals/objectives of the organization.

Reply:
We added a few updates to the ADR and I will show the draft to management to present to our executive director.